tests/jballerina-unit-test/src/test/resources/test-src/types/stream/stream-value.bal

```diff
@@ -621,6 +621,22 @@ function testCastingFromSuperStreamType() {
     var _ = <stream<int, error?>>localEvenNumberStream;
 }
 
+type MyIntStream stream<int>;
+
+function testStreamsTypeAsTypeReference() {
+    var intGenObj = object {
+        int number = 0;
+        public isolated function next() returns record {|int value;|}? {
+            self.number += 1;
+            return {value: self.number};
+        }
+    };
+
+    MyIntStream strm = new (intGenObj);
+    assertValueEquality(1, getGenericRecordValue(strm.next()));
+    assertValueEquality(2, getGenericRecordValue(strm.next()));
+}
```

Comment:
can you add a test with empty initializer
```ballerina
MyIntStream strm = new ();
```

Reply:
Added in [e23a1ba](https://github.com/ballerina-platform/ballerina-lang/pull/38101/commits/e23a1ba0ce174af8153772f5a75c0aa04682edd1)